pkg/api/v1/json.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2021 The TestGrid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+// WriteJSON will write obj to w as JSON, or will write the JSON marshalling error
+func WriteJSON(w http.ResponseWriter, obj interface{}) {
```

Comment:
This is only called once in one function. How about you just inline it? Or at least put it in the same file where it is used. Or at least stop exporting it to everyone.

IMO we should try and minimally pollute:
* The exported package API
* The total package API
*  The directory filename space.

Reply:
I put it in its own file since it seemed like this helper would be useful everywhere: whether we're writing a config, a summary, or whatever, this helper seemed a logical extension to `http.Error`.

I definitely want to use it all over this package, but I can do that without exporting it to everyone.

Reply:
That seems like a good optimization to make once foo.go, bar.go and frobber.go are all calling this method.

Right now though, this is only used by foo.go and thus seems to belong in foo.go